src/components/ResultContainer.tsx

```diff
@@ -194,72 +254,99 @@ function ResultContainer({
                 : parsePipelinesInfoToString(res.available_pipelines),
             ].join('\t')
           );
-        } else {
-          // @ts-expect-error: typescript doesn't know that subject_data is an array when records_protected is false.
-          res.subject_data.forEach((subject) => {
-            tsvRows.push(
-              [
-                res.dataset_name.replace('\n', ' '),
-                res.dataset_portal_uri,
-                res.num_matching_subjects,
-                subject.sub_id,
-                subject.session_id,
-                subject.session_file_path,
-                isFileWithLabels
-                  ? convertURIToLabel('sessionType', subject.session_type)
-                  : subject.session_type,
-                subject.num_matching_phenotypic_sessions,
-                subject.num_matching_imaging_sessions,
-                subject.age,
-                isFileWithLabels ? convertURIToLabel('sex', subject.sex) : subject.sex,
-                isFileWithLabels
-                  ? convertURIToLabel('diagnosis', subject.diagnosis)
-                  : subject.diagnosis,
-                isFileWithLabels
-                  ? convertURIToLabel('assessment', subject.assessment)
-                  : subject.assessment,
-                isFileWithLabels
-                  ? convertURIToLabel('modality', subject.image_modal)
-                  : subject.image_modal?.join(','),
-                isFileWithLabels
-                  ? convertURIToLabel(
-                      'pipeline',
-                      parsePipelinesInfoToString(subject.completed_pipelines).split(',')
-                    )
-                  : parsePipelinesInfoToString(subject.completed_pipelines),
-                isFileWithLabels
-                  ? convertURIToLabel('modality', res.image_modals)
-                  : res.image_modals?.join(','),
-                isFileWithLabels
-                  ? convertURIToLabel(
-                      'pipeline',
-                      parsePipelinesInfoToString(res.available_pipelines).split(',')
-                    )
-                  : parsePipelinesInfoToString(res.available_pipelines),
-              ].join('\t')
-            );
-          });
-        }
+        });
+      }
+    });
+    return tsvRows.join('\n');
+  }
+
+  function constructSubjectsRequestBody(
+    requestBody: DatasetsRequestBody,
+    selectedDatasetUuids: string[],
+    datasetResponses: DatasetsResponse['responses'],
+    nodes: NodeOption[]
+  ): SubjectsRequestBody {
+    // Create a mapping from node_name to node_url (ApiURL)
```

Comment:
I don't think it's a good idea to 

a) put the subject query generation inside the ResultContainer when the responsibility for the dataset query is already in App.tsx - those belong together and should be kept in the same context.
b) Make the subject query depend directly on the dataset query. Even if this makes the code currently simpler, it is harder to understand and creates a coupling that we will have to later disentangle again. It probably also makes testing quite a bit harder to do / reason about.

So instead: 

- let's keep dataset and study request body generation in the same place (ResultContainer may be more appropriate than App.tsx for both)
- make both request body generations independent of each other / read the query parameters independently